### PR TITLE
logging access events to db

### PR DESCRIPTION
**Pull request by dfirebaugh**
- adding some guards if environment variables aren't set
- logging access events in go routines
- renaming db table to `access_events`
- adding `member_counts` table
- adding job to update the monthly stats every day
- adding commits to migrate downs that were missing the commits
- adding better stats from the DB
- can now query based on the month
- removing the old chart functions that were based on payments
- updating swaggerdoc
- removing the `refreshPayments` from the frontend service

### CODE REVIEW

File: api/reports.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		date, err := time.Parse("", month)
+		if err != nil {
+			http.Error(w, "error looking up counts by month - use a valid date", http.StatusNotFound)
+			return
+		}
+		ok(w, makeDistritutionChartByMonth(date, r.store))
+		return
+	}
+
+	var charts []models.PaymentChart
```
Comment (pacna) [inline api/reports.go:30]: `PaymentChart` doesn't make sense anymore. Maybe `ReportChart`

File: docs/swaggerui/swagger.json [hunk 2]
```diff
@@ -380,13 +380,14 @@
         }
       }
     },
-    "/api/member/tier": {
+    "/api/member/stats": {
```
Comment (pacna) [inline docs/swaggerui/swagger.json:383]: Did you make sure to rerun the `gendocs` file? This should no longer exist

File: ui/src/report/services/payment.service.ts [hunk 3]
```diff
@@ -9,15 +9,11 @@ import { PaymentChartResponse } from '../types/api/payment-chart-response';
 
 @Injectable('payment')
```
Comment (pacna) [inline ui/src/report/services/payment.service.ts:10]: `payment` doesn't make any sense anymore. Change this to `report.service.ts` instead

File: ui/src/report/services/payment.service.ts [hunk 4]
```diff
@@ -9,15 +9,11 @@ import { PaymentChartResponse } from '../types/api/payment-chart-response';
...
 export class PaymentService extends HTTPService {
-  private readonly paymentsUrlSegment: string = ENV.api + '/payments';
+  private readonly paymentsUrlSegment: string = ENV.api + '/reports';
 
   getPaymentCharts(): Observable<PaymentChartResponse[]> {
```
Comment (pacna) [inline ui/src/report/services/payment.service.ts:14]: Same with response

File: api/router/reports.go [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+	"memberserver/api/rbac"
+	"net/http"
+)
+
+type ReportsHTTPHandler interface {
+	GetMemberCounts(http.ResponseWriter, *http.Request)
+}
+
+func (r Router) setupReportsRoutes(reports ReportsHTTPHandler, accessControl rbac.AccessControl) {
+	// swagger:route GET /api/member/stats stats searchPaymentChartRequest
```
Comment (pacna) [inline api/router/reports.go:13]: shouldn't this be `/api/reports` instead?

**Review by pacna — CHANGES_REQUESTED** [review]

File: docs/swaggerui/swagger.json (hunk 2 above)
Comment (dfirebaugh) [reply docs/swaggerui/swagger.json:383]: ah, i ran it, but then i didn't run it again

File: api/reports.go [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+
+func makeMemberTrendChart(counts []models.MemberCount) models.ReportChart {
+	var chart models.ReportChart
+	chart.Options.Title = "Membership Trends"
+	chart.Type = "line"
+	chart.Options.CurveType = "function"
+	chart.Options.Legend = "bottom"
+	chart.Cols = []models.ChartCol{{Label: "Month", Type: "string"}, {Label: "Member Count", Type: "number"}}
+
+	sort.Slice(counts, func(i1, i2 int) bool {
```
Comment (pacna) [inline api/reports.go:64]: Why do the sort here? Wouldn't it be better to use sql `orderBy` to sort? A DB sort is more efficient compared to sorting it in the actual codebase

**Review by pacna — APPROVED** [review]

**Pull request merged**